### PR TITLE
Fix dependencies for chomp interface test

### LINKED ISSUE #755: chomp_moveit_test.cpp compilation error: moveit/move_group_interface/move_group_interface.h: No such file or directory
### Description
I get a compilation error when trying to run the unit tests:
```
[ 72%] Built target moveit_ompl_interface
[ 76%] Built target moveit_pick_place_planner
[ 76%] Built target moveit_common_planning_interface_objects
[ 76%] Built target moveit_planning_scene_interface
[ 78%] Built target moveit_move_group_interface
[ 80%] Linking CXX shared library libgtest.so
[ 80%] Built target gtest
Scanning dependencies of target test_cleanup
Scanning dependencies of target gtest_main
Scanning dependencies of target chomp_moveit_test
[ 80%] Building CXX object gtest/CMakeFiles/gtest_main.dir/src/gtest_main.cc.o
[ 80%] Building CXX object moveit/moveit_planners/chomp/chomp_interface/CMakeFiles/chomp_moveit_test.dir/test/chomp_moveit_test.cpp.o
[ 80%] Building CXX object moveit/moveit_ros/planning_interface/test/CMakeFiles/test_cleanup.dir/cleanup.cpp.o
[ 80%] Linking CXX shared library libgtest_main.so
[ 80%] Built target gtest_main
Scanning dependencies of target test_robot_model
Scanning dependencies of target test_transforms
[ 80%] Building CXX object moveit/moveit_core/transforms/CMakeFiles/test_transforms.dir/test/test_transforms.cpp.o
[ 82%] Building CXX object moveit/moveit_core/robot_model/CMakeFiles/test_robot_model.dir/test/test.cpp.o
/tmp/moveit_ws/src/moveit/moveit_planners/chomp/chomp_interface/test/chomp_moveit_test.cpp:40:62: fatal error: moveit/move_group_interface/move_group_interface.h: No such file or directory
compilation terminated.
moveit/moveit_planners/chomp/chomp_interface/CMakeFiles/chomp_moveit_test.dir/build.make:62: recipe for target 'moveit/moveit_planners/chomp/chomp_interface/CMakeFiles/chomp_moveit_test.dir/test/chomp_moveit_test.cpp.o' failed
make[3]: *** [moveit/moveit_planners/chomp/chomp_interface/CMakeFiles/chomp_moveit_test.dir/test/chomp_moveit_test.cpp.o] Error 1
CMakeFiles/Makefile2:15547: recipe for target 'moveit/moveit_planners/chomp/chomp_interface/CMakeFiles/chomp_moveit_test.dir/all' failed
make[2]: *** [moveit/moveit_planners/chomp/chomp_interface/CMakeFiles/chomp_moveit_test.dir/all] Error 2
make[2]: *** Waiting for unfinished jobs....
[ 82%] Linking CXX executable /tmp/moveit_ws/devel/lib/moveit_core/test_transforms
[ 82%] Built target test_transforms
[ 82%] Linking CXX executable /tmp/moveit_ws/devel/lib/moveit_ros_planning_interface/test_cleanup
[ 82%] Built target test_cleanup
[ 82%] Linking CXX executable /tmp/moveit_ws/devel/lib/moveit_core/test_robot_model
[ 82%] Built target test_robot_model
CMakeFiles/Makefile2:113: recipe for target 'CMakeFiles/run_tests.dir/rule' failed
make[1]: *** [CMakeFiles/run_tests.dir/rule] Error 2
Makefile:186: recipe for target 'run_tests' failed
make: *** [run_tests] Error 2
Invoking "make run_tests -j4 -l4" failed
```

### Your environment
* ROS Distro: Kinetic
* OS Version: Ubuntu 16.04
* Source or Binary build? Source
* If binary, which release version?
* If source, which git commit or tag? f50f9fc4025adb7b97f5a8008f7d17627b1be354

### Steps to reproduce

Just do a fresh checkout and run `catkin_make`, then `catkin_make run_tests`:
```
$ mkdir -p /tmp/moveit_ws/src
$ cd /tmp/moveit_ws/src/
$ git clone git@github.com:ros-planning/moveit
$ catkin_init_workspace 
$ cd ..
$ catkin_make
$ catkin_make run_tests
```

### Expected behaviour
All tests run through without error.

### Actual behaviour
I get the following compilation error:
```
/tmp/moveit_ws/src/moveit/moveit_planners/chomp/chomp_interface/test/chomp_moveit_test.cpp:40:62: fatal error: moveit/move_group_interface/move_group_interface.h: No such file or directory
```

### Backtrace or Console output
Full log: https://gist.github.com/stwirth/e00c7459e6b98bda9391c679018712ac

### PR DESCRIPTION
**Pull request by stwirth**
### Description
Fixes #755.

Trivial change: added missing include and dependencies in `CMakeLists.txt` and `package.xml` of chomp_interface.

Without this, `catkin_make run_tests_moveit_planners_chomp` fails with:
```
[100%] Built target moveit_move_group_interface
Scanning dependencies of target chomp_moveit_test
[100%] Building CXX object moveit/moveit_planners/chomp/chomp_interface/CMakeFiles/chomp_moveit_test.dir/test/chomp_moveit_test.cpp.o
/tmp/moveit_ws/src/moveit/moveit_planners/chomp/chomp_interface/test/chomp_moveit_test.cpp:40:62: fatal error: moveit/move_group_interface/move_group_interface.h: No such file or directory
compilation terminated.
moveit/moveit_planners/chomp/chomp_interface/CMakeFiles/chomp_moveit_test.dir/build.make:62: recipe for target 'moveit/moveit_planners/chomp/chomp_interface/CMakeFiles/chomp_moveit_test.dir/test/chomp_moveit_test.cpp.o' failed
make[3]: *** [moveit/moveit_planners/chomp/chomp_interface/CMakeFiles/chomp_moveit_test.dir/test/chomp_moveit_test.cpp.o] Error 1
CMakeFiles/Makefile2:15547: recipe for target 'moveit/moveit_planners/chomp/chomp_interface/CMakeFiles/chomp_moveit_test.dir/all' failed
make[2]: *** [moveit/moveit_planners/chomp/chomp_interface/CMakeFiles/chomp_moveit_test.dir/all] Error 2
CMakeFiles/Makefile2:15483: recipe for target 'moveit/moveit_planners/chomp/chomp_interface/CMakeFiles/run_tests_moveit_planners_chomp.dir/rule' failed
make[1]: *** [moveit/moveit_planners/chomp/chomp_interface/CMakeFiles/run_tests_moveit_planners_chomp.dir/rule] Error 2
Makefile:3930: recipe for target 'run_tests_moveit_planners_chomp' failed
make: *** [run_tests_moveit_planners_chomp] Error 2
Invoking "make run_tests_moveit_planners_chomp -j4 -l4" failed
```

With this change, running `catkin_make run_tests_moveit_planners_chomp` on a clean workspace containing just moveit runs through the test.

### Checklist
- [ ] **Required**: Code is auto formatted using [clang-format](http://moveit.ros.org/documentation/contributing/code)
- [ ] Extended the tutorials / documentation, if necessary [reference](http://moveit.ros.org/documentation/contributing/)
- [ ] Include a screenshot if changing a GUI
- [x] Optional: Created tests, which fail without this PR [reference](http://docs.ros.org/kinetic/api/moveit_tutorials/html/doc/tests.html)
- [ ] Optional: Decide if this should be cherry-picked to other current ROS branches (Indigo, Jade, Kinetic)

[//]: # "You can expect a response from a maintainer within 7 days. If you haven't heard anything by then, feel free to ping the thread. Thank you!"

### CODE REVIEW

File: moveit_planners/chomp/chomp_interface/package.xml [hunk 1]
```diff
@@ -20,7 +20,10 @@
   <build_depend>chomp_motion_planner</build_depend>
   <build_depend>moveit_experimental</build_depend>
 
+  <test_depend>moveit_ros_manipulation</test_depend>
   <test_depend>moveit_ros_planning_interface</test_depend>
+  <test_depend>roscpp</test_depend>
```
Comment (bmagyar) [inline moveit_planners/chomp/chomp_interface/package.xml]: No need to add `roscpp` here, `test-depend` is added on top of the normal `build-depend` list.

**Comment by bmagyar** [comment]
Thanks for the patch! It fixes a couple of issues that only worked for us so far by chance.

**Comment by stwirth** [comment]
@bmagyar fixed!

**Comment by bmagyar** [comment]
Thank you!

**Comment by rhaschke** [comment]
@stwirth Your additions cause warning messages on the [ROS build farm](http://build.ros.org/job/Kdev__moveit__ubuntu_xenial_amd64/260/console) that uses `catkin_make_isolated`:
```
14:40:00 CMake Warning (dev) at CMakeLists.txt:64 (add_dependencies):
14:40:00   Policy CMP0046 is not set: Error on non-existent dependency in
14:40:00   add_dependencies.  Run "cmake --help-policy CMP0046" for policy details.
14:40:00   Use the cmake_policy command to set the policy and suppress this warning.
14:40:00 
14:40:00   The dependency target "move_group" of target "chomp_moveit_test" does not
14:40:00   exist.
```

Can you please fix these?

**Pull request merged**